File: 3251-MaximumAreaOfLongestDiagonalRectangle/3251-MaximumAreaOfLongestDiagonalRectangle.cpp
// Last updated: 04/04/2026, 23:29:40
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        int maxDiagSq = 0;
        int maxArea = 0;
        for (vector<int>& rect : dimensions) {
            int len = rect[0], wid = rect[1];
            int diagSq = len * len + wid * wid;
            int area = len * wid;
            if (diagSq > maxDiagSq) {
                maxDiagSq = diagSq;
                maxArea = area;
            } else if (diagSq == maxDiagSq) {
                maxArea = max(maxArea, area);
            }
        }
        return maxArea;
    }
};

File: 3251-MaximumAreaOfLongestDiagonalRectangle/3251-MaximumAreaOfLongestDiagonalRectangle.java
// Last updated: 04/04/2026, 23:29:46
class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        int ans = 0;
        double maxDiagonal = 0L;
        for(int[] rectangle : dimensions) {
            int len = rectangle[0];
            int wid = rectangle[1];
            double currDiag = Math.sqrt(len*len + wid*wid);
            if(currDiag > maxDiagonal) {
                maxDiagonal = currDiag;
                ans = len * wid;
            } else if (currDiag==maxDiagonal) {
                ans = Math.max(ans,len*wid);
            }
        }
        return ans;
    }
}